h2o-core/src/main/java/water/parser/ParseSetup.java

```diff
@@ -630,14 +598,15 @@ private static final long maxLineLength(byte[] bytes) {
         while(true) {
           line = br.readLine();
           if (line == null) break;
-          maxLineLength = Math.max(line.length(), maxLineLength);
+          int ll = line.length();
+          if (ll > maxLineLength) {
+            maxLineLength = ll;
+          }
         }
-      } catch (IOException e) {
-        return -1;
-      }
+      } catch (IOException ignore) {}
       return maxLineLength;
     }
-    return -1;
```

Comment:
Do you really not need to differentiate an empty file and input made of just empty lines? Why was the -1 there before?